src/Samples/SubModel.Core/Program.fs

```diff
@@ -1,10 +1,66 @@
-module Elmish.WPF.Samples.SubModel.Program
+namespace Elmish.WPF.Samples.SubModel
 
 open System
 open Serilog
 open Serilog.Extensions.Logging
 open Elmish
 open Elmish.WPF
+open System.Runtime.CompilerServices
+
+
+type ViewModelBase2<'model, 'msg>() =
```

Comment:
This type is the first pass attempt. Basically you inherit from this class, call some helpers, then assign the results to the members. This only improved the design-time experience, didn't allow for typed datatemplates, and ended up being quite a bit more lines of code.